anx/provider/loadbalancer/reconciliation/reconciliation_test.go

```diff
@@ -284,6 +284,50 @@ var _ = Describe("reconcile", func() {
 			Expect(recon.servers).To(HaveLen(1))
 		})
 
+		It("should fail to wait for the resources to get ready due to low backoff steps", func() {
+			ctx := context.TODO()
+
+			metrics := metrics.NewProviderMetrics("anexia", "0.0.0-unit-tests")
+
+			// Override reconciliation with only 1 backoff step
+			r, err := New(ctx, apiClient, testClusterName, testLoadBalancerIdentifier, svcUID, externalAddresses, ports, servers, 1, metrics)
+			Expect(err).NotTo(HaveOccurred())
+
+			recon = r.(*reconciliation)
+
+			const waitTime = 5 * time.Second
+
+			time.AfterFunc(waitTime, func() {
+				GinkgoRecover()
+
+				objects := []types.Object{
+					&lbaasv1.Backend{Identifier: httpBackendIdentifier},
+					&lbaasv1.Backend{Identifier: httpsBackendIdentifier},
+					&lbaasv1.Frontend{Identifier: httpFrontendIdentifier},
+					&lbaasv1.Frontend{Identifier: httpsFrontendIdentifier},
+				}
+
+				for _, o := range objects {
+					err := apiClient.Get(context.TODO(), o)
+					Expect(err).NotTo(HaveOccurred())
+
+					switch obj := o.(type) {
+					case *lbaasv1.Backend:
+						obj.HasState.State = lbaasv1.Deployed
+					case *lbaasv1.Frontend:
+						obj.HasState.State = lbaasv1.Deployed
+					}
+
+					err = apiClient.Update(context.TODO(), o)
+					Expect(err).NotTo(HaveOccurred())
+				}
+			})
+
+			err = recon.Reconcile()
+
+			Expect(err).To(HaveOccurred())
```

Comment:
Please always check for specific errors if available. [`wait.ErrWaitTimeout`](https://pkg.go.dev/k8s.io/apimachinery/pkg/util/wait#ExponentialBackoff) in this case.

Reply:
```go
Expect(err).To(BeAssignableToTypeOf(wait.ErrWaitTimeout))
```

does not check for specific errors (`Expect(errors.New("foo")).To(BeAssignableToTypeOf(wait.ErrWaitTimeout))` does also pass)

You can use the following line instead and omit `Expect(err).To(HaveOccurred())`
```go
Expect(err).To(MatchError(wait.ErrWaitTimeout))
```


See https://onsi.github.io/gomega/#matcherrorexpected-interface